keras/applications/imagenet_utils.py

```diff
@@ -28,16 +28,22 @@ def preprocess_input(x, dim_ordering='default'):
     return x
 
 
-def decode_predictions(preds):
+def decode_predictions(preds, top=1):
     global CLASS_INDEX
     assert len(preds.shape) == 2 and preds.shape[1] == 1000
     if CLASS_INDEX is None:
         fpath = get_file('imagenet_class_index.json',
                          CLASS_INDEX_PATH,
                          cache_subdir='models')
         CLASS_INDEX = json.load(open(fpath))
-    indices = np.argmax(preds, axis=-1)
+    
+    if top > 1:
+        indices = (-preds).argsort()[:top]
+        top_preds = indices[0, :top]
+    else:
+        top_preds = np.argmax(preds, axis=-1)
+    
     results = []
-    for i in indices:
+    for i in top_preds:
```

Comment:
This function is supposed to work for *batches* of predictions, not a single prediction. You seem to have written your code under the assumption that there was a single sample being considered.